### PR TITLE
Avoid duplicate markdown sanitization

### PR DESCRIPTION
**Pull request by DonJayamanne**
For https://github.com/microsoft/vscode/issues/210528

### CODE REVIEW

File: src/vs/base/browser/markdownRenderer.ts [hunk 1]
```diff
@@ -306,7 +309,9 @@ export function renderMarkdown(markdown: IMarkdownString, options: MarkdownRende
 			}
 		});
 
-	element.innerHTML = sanitizeRenderedMarkdown(markdown, markdownHtmlDoc.body.innerHTML) as unknown as string;
+	const html = markdownHtmlDoc.body.innerHTML;
+	const trustedhtml = ttPolicy ? ttPolicy.createHTML(html) : html;
```
Comment (DonJayamanne) [inline src/vs/base/browser/markdownRenderer.ts:313]: Given the fact that we already sanitize the markdown in line 267, i think we can remove the second level of sanitization.

However I could be completely wrong, hence feel free to close this issue.

File: src/vs/base/test/browser/markdownRenderer.test.ts [hunk 2]
```diff
@@ -186,7 +186,7 @@ suite('MarkdownRenderer', () => {
 </thead>
 <tbody><tr>
 <td><span class="codicon codicon-zap"></span></td>
-<td><a data-href="#link" href="" title="#link" draggable="false"><span class="codicon codicon-zap"></span>-link</a></td>
+<td><a draggable="false" title="#link" href="" data-href="#link"><span class="codicon codicon-zap"></span>-link</a></td>
```
Comment (DonJayamanne) [inline src/vs/base/test/browser/markdownRenderer.test.ts:189]: Only changed the order of the attributes, else the HTML is identical

File: src/vs/base/browser/markdownRenderer.ts (hunk 1 above)
Comment (mjbvz) [reply src/vs/base/browser/markdownRenderer.ts:313]: Possibly but if anything I'd prefer removing the first sanitize call. This is the final pass so if we remove it there's a risk that the post processing steps that happen after the first sanitize call but before this one could introduce a security issue

**Comment by mjbvz** [comment]
Also would like to get a repo. In my testing of the linked issue, the call to sanitize takes under 1ms. Even if we don't call sanitize twice, it sounds like we'd still be taking 150ms or so which isn't acceptable

**Comment by DonJayamanne** [comment]
> n my testing of the linked issue, the call to sanitize takes under 1ms. Even if we don't call sanitize twice, it sounds like we'd still be taking 150ms or so which isn't acceptable

Sometimes I have seen this taking 300ms when working on notebooks, we use markdown parser to build outlines and that ends up parsing markdown which results in 300ms sometimes.
So, this is definitely happening in other places too, but not 100% of the time.
The issue does definitely repro for me, please let me know if CPU profiles will help and what else information you need.

**Pull request closed without merging**